config/routes.rb

```diff
@@ -3,4 +3,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:user_id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
```

Comment:
- [ ] [Optional] It would be better professionally if you add your routes like that:
```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
end

```

Reply:
@Omar-Muhamad Great! I will apply it, thank you very much for the advice!